drivers/include/net/netdev.h

```diff
@@ -236,7 +236,14 @@ typedef enum {
     NETDEV_EVENT_RX_COMPLETE,               /**< finished receiving a frame */
     NETDEV_EVENT_TX_STARTED,                /**< started to transfer a frame */
     NETDEV_EVENT_TX_COMPLETE,               /**< transfer frame complete */
-    NETDEV_EVENT_TX_COMPLETE_DATA_PENDING,  /**< transfer frame complete and data pending flag */
+    /**
+     * @brief   transfer frame complete and data pending flag 
+     *
+     * @deprecated  Issue an NETDEV_EVENT_TX_COMPLETE event instead and pass
+     *              the data pending info in netdev_driver_t::confirm_send
+     *              via the `info` parameter
+     */
+    NETDEV_EVENT_TX_COMPLETE_DATA_PENDING,
     NETDEV_EVENT_TX_NOACK,                  /**< ACK requested but not received */
```

Comment:
Shoudln't `NOACK` and `TX_MEDIUM_BUSY` be included as well? They are resolved at the same time

Reply:
Done. I changed the return values to have `-EGAIN` in `confirm_send()` when called to early. This matches the semantics of the error code better. And this allows to return `-EBUSY` when the medium was busy; which is consistent with the interpretation of the same error code issued by `send()`.